lnst/RecipeCommon/Perf/Measurements/XDPBenchMeasurement.py

```diff
@@ -0,0 +1,180 @@
+            for flow in self._running_measurements:
+                client_job = flow.client_job.what
+                flow.client_job.wait(timeout=client_job.runtime_estimate())
+                flow.server_job.wait(timeout=5)
+        finally:
+            for flow in self._running_measurements:
+                flow.server_job.kill()
+                flow.client_job.kill()
+        self._finished_measurements = self._running_measurements
+        self._running_measurements = []
+
+    def collect_results(self):
+        test_flows = self._finished_measurements
+
+        results = []
+        for test_flow in test_flows:
+            flow_results = XDPBenchMeasurementResults(
+                measurement=self,
+                flow=test_flow.flow,
+                warmup_duration=test_flow.flow.warmup_duration,
+            )
+            flow_results.generator_results = self._parse_generator_results(
+                test_flow.client_job,
+                test_flow.flow.duration + test_flow.flow.warmup_duration * 2,
+            )
+            flow_results.receiver_results = self._parse_receiver_results(
+                test_flow.server_job
+            )
+
+            results.append(flow_results)
+        return results
+
+    def _parse_generator_results(self, job: Job, duration: int):
+        results = ParallelPerfResult()  # container for multiple instances of pktgen
+
+        for device, raw_result in job.result.items():
+            result = SequentialPerfResult()  # instance (device) of pktgen
+
+            cumulative_sample = PerfInterval(
+                raw_result["result"]["packets"],
+                raw_result["result"]["duration"],
+                "packets",
+                raw_result["current"]["started"],
+            )
+            result.extend(cumulative_sample.resample(duration))
+            results.append(result)
+
+        return results
+
+    def _parse_receiver_results(self, job: Job):
+        result = ParallelPerfResult()  # just a placeholder to keep data structure same as other Measurements
+        results = SequentialPerfResult()  # single instance of xdp-bench
+
+        for i, sample in enumerate(job.result):
+            results.append(PerfInterval(sample["rx"], 1, "packets", i))
+
+        result.append(results)
+
+        return result
+
+    def _aggregate_flows(self, old_flow, new_flow):
+        if old_flow is not None and old_flow.flow is not new_flow.flow:
+            raise MeasurementError("Aggregating incompatible Flows")
+
+        new_result = AggregatedXDPBenchMeasurementResults(measurement=self, flow=new_flow.flow)
+
+        new_result.add_results(old_flow)
+        new_result.add_results(new_flow)
+        return new_result
+
+    @classmethod
+    def _report_flow_results(cls, recipe, flow_results):
+        generator = flow_results.generator_results
+        receiver = flow_results.receiver_results
+       
+        desc = []
+        desc.append(str(flow_results.flow))
```

Comment:
for the other measurement classes we've moved the generation of the description into the `Result` classes: https://github.com/LNST-project/lnst/blob/master/lnst/RecipeCommon/Perf/Measurements/Results/FlowMeasurementResults.py#L107

Reply:
thanks, missed that, moved.